editor/doc_tools.cpp

```diff
@@ -96,221 +124,96 @@ void DocTools::merge_from(const DocTools &p_data) {
-								arg_used.write[l] = true;
-								break;
-							}
-						}
-					}
-					bool not_the_same = false;
-					for (int l = 0; l < arg_count; ++l) {
-						if (!arg_used[l]) { // at least one of the arguments was different
-							not_the_same = true;
-						}
-					}
-					if (not_the_same) {
-						continue;
-					}
-				}
+		merge_methods(c.constructors, cf.constructors);
 
-				const DocData::MethodDoc &mf = cf.constructors[j];
+		merge_methods(c.methods, cf.methods);
 
-				m.description = mf.description;
-				m.is_deprecated = mf.is_deprecated;
-				m.deprecated_message = mf.deprecated_message;
-				m.is_experimental = mf.is_experimental;
-				m.experimental_message = mf.experimental_message;
-				break;
-			}
-		}
+		merge_methods(c.signals, cf.signals);
 
-		for (int i = 0; i < c.methods.size(); i++) {
-			DocData::MethodDoc &m = c.methods.write[i];
+		// Perform ordered merge on names.
+		{
+			Vector<DocData::ConstantDoc>::Iterator this_itr = c.constants.begin();
+			Vector<DocData::ConstantDoc>::ConstIterator other_itr = cf.constants.begin();
 
-			for (int j = 0; j < cf.methods.size(); j++) {
-				if (cf.methods[j].name != m.name) {
-					continue;
+			while (this_itr != c.constants.end() && other_itr != cf.constants.end()) {
+				while (this_itr != c.constants.end() && this_itr->name < other_itr->name) {
+					++this_itr;
+				}
+				while (other_itr != cf.constants.end() && this_itr->name > other_itr->name) {
+					++other_itr;
+				}
+				if (this_itr == c.constants.end() || other_itr == cf.constants.end()) {
+					break;
 				}
 
-				const DocData::MethodDoc &mf = cf.methods[j];
+				this_itr->description = other_itr->description;
+				this_itr->is_deprecated = other_itr->is_deprecated;
+				this_itr->deprecated_message = other_itr->deprecated_message;
+				this_itr->is_experimental = other_itr->is_experimental;
+				this_itr->experimental_message = other_itr->experimental_message;
+				this_itr->keywords = other_itr->keywords;
 
-				m.description = mf.description;
-				m.is_deprecated = mf.is_deprecated;
-				m.deprecated_message = mf.deprecated_message;
-				m.is_experimental = mf.is_experimental;
-				m.experimental_message = mf.experimental_message;
-				m.keywords = mf.keywords;
-				break;
+				++this_itr;
+				++other_itr;
 			}
 		}
 
-		for (int i = 0; i < c.signals.size(); i++) {
-			DocData::MethodDoc &m = c.signals.write[i];
+		merge_methods(c.annotations, cf.annotations);
 
-			for (int j = 0; j < cf.signals.size(); j++) {
-				if (cf.signals[j].name != m.name) {
-					continue;
-				}
-				const DocData::MethodDoc &mf = cf.signals[j];
-
-				m.description = mf.description;
-				m.is_deprecated = mf.is_deprecated;
-				m.deprecated_message = mf.deprecated_message;
-				m.is_experimental = mf.is_experimental;
-				m.experimental_message = mf.experimental_message;
-				m.keywords = mf.keywords;
-				break;
-			}
-		}
+		// Perform ordered merge on names.
+		{
```

Comment:
I think it would be great to do the same thing as `merge_methods` in this PR, and group the other categories as well into their own method. Being left out like this in favour of code regions makes the code look very unwieldy and disorganized. Especially more with short and repeating variable names.

Reply:
I can put them into their own single methods if desired